server/src/server.ts

```diff
@@ -179,8 +183,10 @@ function openProjectFile(workspacePath: string, fileUri: string): boolean {
         .readFile(fileUri)
         .then((_file) => {
             const fileContent = _file.toString();
-            parseFile(`file://${workspacePath}/${fileUri}`, fileContent);
-            if (!fileContent.trim().startsWith('_FILE_TYPE_')) {
+            const isTypeF = fileContent.search(FILE_TYPE_F) !== -1;
```

Comment:
I’m a bit worried about the fact that a “startsWith” turned into a double full regex search. No way to restrain that a bit? Our parsing step is already costly.

Reply:
Well, I thought about it a little bit too. But:
* since we cannot be sure of how many spaces there could be between `_FILE_TYPE_ and `F` or `M`, I thought it would be better with a regex. 
* note that the second one is done only when the first one failed.
* in the precedent version, we were doing a `trim` first, maybe this is costly too?
* the pattern we are looking for is only at the start of the string, I suppose it will fail very fast if possible and not be that costly?

Reply:
Use numbered lists to make it easier to answer you xD

. Spaces: Sure.
. Only if failed: Yeah but there’s a large majority of files are neither F or M, right? Oh but maybe we only have .kao files, here? Or not?
. Trim: No, I think trim is trivial.
. `^`: I missed that. That makes the other points kinda irrelevant. OK for this stuff, I suppose.

Reply:
2. You're right, damn :/